src/App.tsx

```diff
@@ -1,61 +1,141 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { CompletedTodo, HandleEvent, TypeTodo } from './type';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUser = (todo: TypeTodo) => {
+  return usersFromServer.find(user => user.id === todo.userId);
+};
+
+const todos: CompletedTodo[] = todosFromServer.map(todo => ({
+  todo,
+  user: getUser(todo),
+}));
+
+const maxIndex = todosFromServer
+  .reduce((max, todo): number => (max < todo.id ? todo.id : max), 0);
+
+const createTodo = () => {
+  let currentId = maxIndex;
+
+  const Todo = (title: string, userId: number): TypeTodo => {
+    currentId += 1;
+
+    return {
+      title,
+      userId,
+      id: currentId,
+      completed: false,
+    };
+  };
+
+  return Todo;
+};
 
 export const App = () => {
+  const [stateTodos, setTodos] = useState(todos);
+  const [chooseUser, setUser] = useState('empty');
+  const [newTitle, setTitle] = useState('');
+  const [isErrorTitle, setErrorTitle] = useState(false);
+  const [isErrorUser, setErrorUser] = useState(false);
```

Comment:
```suggestion
  const [todos, setTodos] = useState(todos);
  const [chooseUser, setChooseUser] = useState('empty');
  const [newTitle, setNewTitle] = useState('');
  const [isErrorTitle, setIsErrorTitle] = useState(false);
  const [isErrorUser, setIsErrorUser] = useState(false);
```
Благодаря таким названиям будет понятно, какие именно значения и функции связаны. Старайся называть так: [example, setExample]